crates/context_aware_config/src/api/context/handlers.rs

```diff
@@ -247,25 +252,23 @@ fn get_put_resp(ctx: Context) -> PutResp {
 
 fn put(
     req: Json<PutReq>,
-    conn: &mut PooledConnection<ConnectionManager<PgConnection>>,
-    already_under_txn: bool,
+    transaction_conn: &mut PooledConnection<ConnectionManager<PgConnection>>,
     user: &User,
 ) -> superposition::Result<PutResp> {
     use contexts::dsl::contexts;
-    let new_ctx = create_ctx_from_put_req(req, conn, user)?;
+    let new_ctx = create_ctx_from_put_req(req, transaction_conn, user)?;
 
-    if already_under_txn {
-        diesel::sql_query("SAVEPOINT put_ctx_savepoint").execute(conn)?;
-    }
-    let insert = diesel::insert_into(contexts).values(&new_ctx).execute(conn);
+    diesel::sql_query("SAVEPOINT put_ctx_savepoint").execute(transaction_conn)?;
```

Comment:
Why did we change this? Are we saying that all calls to this function are within a transaction. If not, this code will fail.

Reply:
yes @Datron all call to this function are within a transaction

Reply:
Can you put this back as it was? Since in future we might move away from doing snapshot in transaction and go ahead with some other method.